samples/nrf9160/lwm2m_client/src/main.c

```diff
@@ -356,11 +356,44 @@ static void rd_client_event(struct lwm2m_ctx *client,
 		break;
 
 	case LWM2M_RD_CLIENT_EVENT_NETWORK_ERROR:
-		LOG_ERR("LwM2M engine reported a network erorr.");
+		LOG_ERR("LwM2M engine reported a network error.");
+		k_sem_give(&lwm2m_restart);
 		break;
 	}
 }
 
+static void modem_connect(void)
+{
+	int ret;
+
+	do {
+		LOG_INF("Connecting to LTE network.");
+		LOG_INF("This may take several minutes.");
+		ui_led_set_pattern(UI_LTE_CONNECTING);
+
+		ret = lte_lc_connect();
+		if (ret < 0) {
+			LOG_WRN("Failed to establish LTE connection (%d).",
+				ret);
+			LOG_WRN("Will retry in a minute.");
+			lte_lc_offline();
+			k_sleep(K_SECONDS(60));
+		} else {
+			LOG_INF("Connected to LTE network");
+			ui_led_set_pattern(UI_LTE_CONNECTED);
+		}
+	} while (ret < 0);
+
+#if defined(CONFIG_LWM2M_QUEUE_MODE_ENABLED)
+	ret = lte_lc_psm_req(true);
```

Comment:
I'd put the PSM request before do-while loop. It saves an extra TAU request after connection is established and in some networks that TAU request may be rejected for an extended period of time (several minutes with repeated TAU requests and rejections going back and forth).

Reply:
Thanks for the tip, suggestion applied.